### PR TITLE
Delete a person's transactions when deleting a person

### LINKED ISSUE #142: Relationship between `Person` and `Transaction`
We have not formally defined referential integrity between `Person` and `Transaction`. When a person is deleted, what should happen to their transactions?

1. Should transactions immediately be deleted when a person is deleted? This matters because it will affect the way `LogicManager` needs to be restructured.
2. If not, should we do it such that the transaction storage simply ignores transactions with hanging references?

### PR DESCRIPTION
**Pull request by seanlim**
Closes #142 

Ensure that when a person is deleted, all their associated transactions are also removed.

## Key Changes:

1. Introduced `deleteTransactionsOfPersonId` method to remove transactions linked to a specific person ID.
2. Updated `DeleteCommand` to call `deleteTransactionsOfPersonId` after deleting a person.
3. Refactored `ModelManager` to `AddressBookModelManager` and `TransactionBookModelManager` for clarity.
4. Modified tests to accommodate the new behavior and refactored class names.

### CODE REVIEW

File: src/main/java/spleetwaise/transaction/model/TransactionBookModelManager.java [hunk 1]
```diff
@@ -8,22 +8,23 @@
...
 import spleetwaise.transaction.model.transaction.Transaction;
 
 /**
  * Represents the in-memory model of the transaction data.
  */
-public class ModelManager implements TransactionBookModel {
+public class TransactionBookModelManager implements TransactionBookModel {
 
-    private static final Logger logger = LogsCenter.getLogger(spleetwaise.address.model.ModelManager.class);
+    private static final Logger logger = LogsCenter.getLogger(AddressBookModelManager.class);
```
Comment (rollingpencil) [inline src/main/java/spleetwaise/transaction/model/TransactionBookModelManager.java:19]:
```suggestion
    private static final Logger logger = LogsCenter.getLogger(TransactionBookModelManager.class);
```
Typo for class name

File: src/main/java/spleetwaise/transaction/model/TransactionBookModelManager.java [hunk 2]
```diff
@@ -8,22 +8,23 @@
...
 
     private final TransactionBook transactionBook;
     private final FilteredList<Transaction> filteredTransactions;
 
     /**
-     * Initializes a ModelManager with the given transactionBook.
+     * Initializes a AddressBookModelManager with the given transactionBook.
```
Comment (rollingpencil) [inline src/main/java/spleetwaise/transaction/model/TransactionBookModelManager.java:25]:
```suggestion
     * Initializes a TransactionBookModelManager with the given transactionBook.
```

File: src/main/java/spleetwaise/transaction/model/TransactionBookModelManager.java [hunk 3]
```diff
@@ -33,9 +34,9 @@ public ModelManager(TransactionBook transactionBook) {
     }
 
     /**
-     * Initializes a ModelManager with the readonly transactionBook.
+     * Initializes a AddressBookModelManager with the readonly transactionBook.
```
Comment (rollingpencil) [inline src/main/java/spleetwaise/transaction/model/TransactionBookModelManager.java:37]:
```suggestion
     * Initializes a TransactionBookModelManager with the readonly transactionBook.
```

**Review by rollingpencil — APPROVED** [review]
Looks good to me! Just a few typos I found and it can be merged afterwards!

**Review by SeeYangZhi — APPROVED** [review]
This pull request addresses issue #142 by implementing functionality to delete all transactions associated with a person when that person is deleted. The key changes include:

1. New Method: Introduced deleteTransactionsOfPersonId to remove transactions linked to a specific person ID.
2. Command Update: Modified the DeleteCommand to call deleteTransactionsOfPersonId after deleting a person.
3. Refactoring: Renamed ModelManager to AddressBookModelManager and TransactionBookModelManager for better clarity.
4. Test Adjustments: Updated tests to reflect the new behavior and refactored class names accordingly.

These changes ensure that the system maintains data integrity by automatically cleaning up related transactions when a person is removed from the address book.

**Pull request merged**